File: rxrecycler-adapter/src/main/java/com/minimize/android/rxrecycleradapter/TypesViewHolder.java
package com.minimize.android.rxrecycleradapter;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by ahmedrizwan on 13/12/2015.
 */
public class TypesViewHolder<T> extends RecyclerView.ViewHolder {
    private ViewDataBinding mViewDataBinding;

    public T getItem() {
        return mItem;
    }

    public ViewDataBinding getViewDataBinding() {
        return mViewDataBinding;
    }

    private T mItem;

    protected void setItem(final T item) {
        mItem = item;
    }

    public TypesViewHolder(final View itemView) {
        super(itemView);
        mViewDataBinding = DataBindingUtil.bind(itemView);
    }
}

File: rxrecycler-adapter/src/main/java/com/minimize/android/rxrecycleradapter/TypesViewHolder.kt
package com.minimize.android.rxrecycleradapter

import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by ahmedrizwan on 13/12/2015.
 */
open class TypesViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val viewDataBinding: ViewDataBinding? = DataBindingUtil.bind(itemView)

    var item: T? = null
}
